using-mobile-extensions/adobe-journey-optimizer-decisioning/api-reference.md

```diff
@@ -0,0 +1,1005 @@
+                                                                     itemCount: 2];
+AEPDecisionScope* decisionScope2 = [[AEPDecisionScope alloc] initWithName: @"myScope"];
+
+[AEPMobileOptimize getPropositions: @[decisionScope1, decisionScope2] 
+                        completion: ^(NSDictionary<AEPDecisionScope*, AEPProposition*>* propositionsDict, NSError* error) {
+  if (error != nil) {
+    // handle error   
+    return;
+  }
+
+  AEPProposition* proposition1 = propositionsDict[decisionScope1];
+  // read proposition1 offers
+
+  AEPProposition* proposition2 = propositionsDict[decisionScope2];
+  // read proposition2 offers
+}];
+```
+
+{% endtab %}
+{% endtabs %}
+
+## onPropositionsUpdate
+
+This API registers a permanent callback which is invoked whenever the Edge extension dispatches a response Event received from the Experience Edge Network upon a personalization query. The personalization query requests can be triggered by the `updatePropositions` API, Edge extension `sendEvent` API or launch consequence rules.
+
+{% tabs %}
+{% tab title="Android" %}
+### Java
+
+#### Syntax
+
+```java
+public static void onPropositionsUpdate(final AdobeCallback<Map<DecisionScope, Proposition>> callback)
+```
+
+* _callback_ `call` method is invoked with propositions map of type `Map<DecisionScope, Proposition>`. If the callback is an instance of `AdobeCallbackWithError`, and if the operation times out or an error occurs in retrieving propositions, the `fail` method is invoked with the appropriate `AdobeError`.
+
+#### Example
+
+```java
+Optimize.onPropositionsUpdate(new AdobeCallbackWithError<Map<DecisionScope, Proposition>>() {
+    @Override
+    public void fail(final AdobeError adobeError) {
+        // handle error
+    }
+
+    @Override
+    public void call(final Map<DecisionScope, Proposition> propositionsMap) {
+        if (propositionsMap != null && !propositionsMap.isEmpty()) {
+            // handle propositions
+        }
+    }
+});
+```
+{% endtab %}
+{% tab title="iOS (AEP 3.x)" %}
+### Swift
+
+#### Syntax
+
+```swift
+static func onPropositionsUpdate(perform action: @escaping ([DecisionScope: Proposition]?) -> Void)
+```
+
+* _action_ is invoked with propositions dictionary of type `[DecisionScope: Proposition]`.
+
+#### Example
+
+```swift
+Optimize.onPropositionsUpdate { propositionsDict in
+  if let propositionsDict = propositionsDict {
+    // handle propositions
+  }
+}
+```
+
+### Objective-C
+
+#### Syntax
+
+```objc
++ (void) onPropositionsUpdate: (void (^ _Nonnull)(NSDictionary<AEPDecisionScope*, AEPProposition*>* _Nullable)) action;
+```
+
+* _action_ is invoked with propositions dictionary of type `NSDictionary<AEPDecisionScope*, AEPProposition*>`.
+
+#### Example
+
+```objc
+[AEPMobileOptimize onPropositionsUpdate: ^(NSDictionary<AEPDecisionScope*, AEPProposition*>* propositionsDict) {
+  // handle propositions
+}];
+```
+
+{% endtab %}
+{% endtabs %}
+
+## registerExtension(s)
```

Comment:
A question more out of curiosity than something that's an issue - is there a reason that it's extension on Android and extension**s** on iOS? IE: Why is it not the same between the two platforms?

Reply:
That's because it was introduced during Swift rewrite, we might introduce it on Android too with the next major update.